### ISSUE TITLE
Update dependencies!

### ISSUE THREAD

**Issue by aadrian** [ContainX/openstack4j#985]
Please update dependencies,

According to https://www.versioneye.com/

most dependencies of openstack4j are outdated.

Thank you.

**Comment by auhlig** [comment]
This is an open-source project driven by the community. You're very welcome to submit a PR and help updating our dependencies and enhancing OpenStack4j @aadrian.

**Comment by aadrian** [comment]
@auhlig gladly, but why do different modules have different versions?

**Comment by auhlig** [comment]
Cool. If you were able to identify specific issues that should be addressed I would be glad if you could submit a PR for those.

**Comment by auhlig** [comment]
Please open more specific issues if you started working in something.

**Comment by aadrian** [comment]
@auhlig I started updating them, but found some strange issues that don't seem to make sense or at least are not explained in the docs, hence my questions:
 - why is there a ```openstack4j-core-2.0.0.jar``` instead of ```openstack4j-core-3.0.4-SNAPSHOT.jar ``` in ```openstack4j\core\target``` since there's a ```openstack4j-core-3.0.4-SNAPSHOT-withdeps.jar``` in that directory too ?
 - many sub-modules seem to have this.  What is the official version anyway? ```2.0.0``` or ```3.0.4-SNAPSHOT``` (since ```3.0.2``` was released in November)?
 - what is the official Java version for the latest version of the library?
